evap/results/views.py

```diff
@@ -199,6 +209,25 @@ def evaluation_detail(request, semester_id, evaluation_id):
 
     is_responsible_or_contributor_or_delegate = evaluation.is_user_responsible_or_contributor_or_delegate(view_as_user)
 
+    contributor_textanswers = view_as_user.is_reviewer or any(
+        evaluation.is_user_contributor(user) for user in represented_users
+    )
+    contributor_personal = evaluation.is_user_contributor(view_as_user)
+
+    user_represents_responsible = any(user in represented_users for user in evaluation.course.responsibles.all())
+    user_represents_general_visibilty_contributor = any(
+        evaluation.is_user_contributor(user)
+        and evaluation.contributions.filter(
+            contributor=user,
+            textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+        ).exists()
+        for user in represented_users
+    )
```

Comment:
Semantically correct and nice code, _but_: We also need to worry about performance here. `is_user_contributor` hits the database once, the `.exists()` query hits the database once, all of this happens `n` times for `n` represented users. Can we maybe express this as a single queryset instead?

(same thing applies to the other queries above as well)